File: go/assets/decrypt.go
package assets

import (
	"fmt"
	"strconv"
)

func decryptTable() map[string]int {
	e2i := make(map[string]int)
	currentNum := 0
	emojis := getEmojis()

	for _, emoji := range emojis {
		e2i[string(emoji)] = currentNum
		currentNum++
	}

	return e2i
}

func decryptStream(numbers []int) (stream string) {
	for _, number := range numbers {
		stream += fmt.Sprintf("%0.07b", int(number))
	}
	return
}

func decryptSplit(stream string) (bins []string) {

	current := ""

	for _, bin := range stream {
		if len(current) < 8 {
			current += string(bin)
		} else {
			bins = append(bins, current)
			current = string(bin)
		}
	}
	return
}

func Decode(message string) string {

	e2i := decryptTable()
	var numbers []int

	for _, emoji := range message {
		numbers = append(numbers, e2i[string(emoji)])
	}
	binaryStream := decryptStream(numbers)
	bins := decryptSplit(binaryStream)

	decryptedMessage := ""

	for _, bin := range bins {
		i, _ := strconv.ParseInt(string(bin), 2, 64)
		decryptedMessage += string(int(i))
	}

	return decryptedMessage
}

File: docs/js/decrypt.js
function getDecryptTable() {
    let emojis = "😀😃😄😁😆😅🤣😂🙂🙃😉😊😇🥰😍🤩😘😗😚😙😋😛😜🤪😝🤑🤗🤭🤫🤔🤐🤨😐😑😶😏😒🙄😬🤥😌😔😪🤤😴😷🤒🤕🤢🤮🤧🥵🥶🥴😵🤯🤠🥳😎🤓🧐😕😟🙁😮😯😲😳🥺😦😧😨😰😥😢😭😱😖😣😞😓😩😫🥱😤😡😠🤬😈👿💀💩🤡👹👺👻👽👾🤖😺😸😹😻😼😽🙀😿😾💋👋🤚🖐✋🖖👌🤏🤞🤟🤘🤙👈👉👆🖕👇👍👎✊"
    let e2i = new Map()
    let index = 0
    for(let emoji of emojis) {
        e2i.set(emoji, index)
        index++
    }
    return e2i
}

function getBin(number) {
    let bin = number.toString(2)
    while(bin.length < 7) {
        bin = "0" + bin
    }
    return bin
}

function split(binaryStream) {
    let bins = []
    let current = ""
    for(let bin of binaryStream) {
        if(current.length < 8) {
            current += bin
        } else {
            bins.push(current)
            current = bin
        }
    }
    return bins
}

function decrypt(message) {
    let e2i = getDecryptTable()
    let binaryStream = ""
    for(let emoji of message) {
        let number = e2i.get(emoji)
        binaryStream += getBin(number)
    }
    let bytes = split(binaryStream)
    let text = ""
    for(let byte of bytes) {
        number = parseInt(byte, 2)
        text += String.fromCharCode(number)
    }
    return text
}